e3sm_diags/driver/zonal_mean_xy_driver.py

```diff
@@ -1,262 +1,316 @@
-                            mv1_zonal,
-                            diff,
-                            {},
-                            parameter,
-                        )
-                        utils.general.save_ncfiles(
-                            parameter.current_set,
-                            mv1_zonal,
-                            mv2_zonal,
-                            diff,
-                            parameter,
-                        )
-
-            # For variables without a z-axis.
-            elif mv1.getLevel() is None and mv2.getLevel() is None:
-                for region in regions:
-                    logger.info(f"Selected region: {region}")
-                    mv1_zonal = cdutil.averager(mv1, axis="x")
-                    mv2_zonal = cdutil.averager(mv2, axis="x")
-
-                    mv1_reg, mv2_reg = regrid_to_lower_res_1d(mv1_zonal, mv2_zonal)
-
-                    diff = mv1_reg - mv2_reg
-
-                    parameter.output_file = "-".join([ref_name, var, season, region])
-                    parameter.main_title = str(" ".join([var, season, region]))
-
-                    parameter.var_region = region
-
-                    plot(
-                        parameter.current_set,
-                        mv2_zonal,
-                        mv1_zonal,
-                        diff,
-                        {},
-                        parameter,
-                    )
-                    utils.general.save_ncfiles(
-                        parameter.current_set,
-                        mv1_zonal,
-                        mv2_zonal,
-                        diff,
-                        parameter,
-                    )
-
-            else:
+            elif is_dims_diff:
                 raise RuntimeError(
                     "Dimensions of the two variables are different. Aborting."
                 )
 
     return parameter
+
+
+def _run_diags_2d(
+    parameter: CoreParameter,
+    ds_test: xr.Dataset,
+    ds_ref: xr.Dataset,
+    season: str,
+    regions: List[str],
+    var_key: str,
+    ref_name: str,
+):
+    """Run diagnostics on a 2D variable.
+
+    This function gets the variable's metrics by region, then saves the
+    metrics, metric plots, and data (optional, `CoreParameter.save_netcdf`).
+
+    Parameters
+    ----------
+    parameter : CoreParameter
+        The parameter object.
+    ds_test : xr.Dataset
+        The dataset containing the test variable.
+    ds_ref : xr.Dataset
+        The dataset containing the ref variable. If this is a model-only run
+        then it will be the same dataset as ``ds_test``.
+    season : str
+        The season.
+    regions : List[str]
+        The list of regions.
+    var_key : str
+        The key of the variable.
+    ref_name : str
+        The reference name.
+    """
+    for region in regions:
+        parameter._set_param_output_attrs(var_key, season, region, ref_name, ilev=None)
+        # TODO: Fix type annotation
```

Comment:
I added a few TODO: statements to fix the `mypy` type annotation issues.

Most of them come up because the return type for `_calculate_zonal_mean()`  hasn't been updated.